driver-core/src/main/com/mongodb/operation/FindAndDeleteOperation.java

```diff
@@ -47,6 +57,7 @@
     private BsonDocument projection;
     private BsonDocument sort;
     private long maxTimeMS;
+    private WriteConcern writeConcern;
```

Comment:
I'd like these classes to be more consistent with BaseWriteOperation. To do so would require the following changes:

* Add a new constructor that takes a non-null WriteConcern parameter
* Deprecate the existing constructor  (so as not to break binary compatibility for any users of core)
* Remove the new writeConcern setter method
* For backwards compatibility, default the writeConcern field to ACKNOWLEDGED
* When creating the command document, instead of checking for null, check for WriteConcern.isServerDefault()  (as in BaseWriteCommandMessage)